Makefile.uk

```diff
@@ -0,0 +1,105 @@
+
+################################################################################
+# Helpers
+##############################################################################
+LIBPYTHON_SHAPELY_SRC = $(LIBPYTHON_SHAPELY_ORIGIN)/$(LIBPYTHON_SHAPELY_DIRNAME)
+
+################################################################################
+# Library includes
+##############################################################################
+LIBPYTHON_SHAPELY_CINCLUDES += -I$(LIBPYTHON_SHAPELY_SRC)/src
+
+################################################################################
+# Global flags
+##############################################################################
+LIBPYTHON_SHAPELY_FLAGS-y += -fwrapv
+LIBPYTHON_SHAPELY_FLAGS-y += -Wno-unused-result
+LIBPYTHON_SHAPELY_FLAGS-y += -Wsign-compare
+LIBPYTHON_SHAPELY_FLAGS-$(call have_clang) += -Wno-documentation
+
+LIBPYTHON_SHAPELY_CFLAGS += $(LIBPYTHON_SHAPELY_FLAGS-y)
+
+################################################################################
+# Library sources
+##############################################################################
+# Upstream C sources
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/c_api.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/coords.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/geos.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/lib.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/pygeom.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/strtree.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/ufuncs.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_SRC)/src/vector.c
+
+# Cythonized sources
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_BASE)/generated/shapely/_geometry_helpers.c
+LIBPYTHON_SHAPELY_SRCS-y += $(LIBPYTHON_SHAPELY_BASE)/generated/shapely/_geos.c
+
+################################################################################
+# Root filesystem
+##############################################################################
+
+ifeq ($(CONFIG_LIBPYTHON_SHAPELY),y)
+# Install into main python rootfs & cleanup non-python files
+$(PYTHON_ROOTFS)/.shapely_done: $(PYTHON_ROOTFS)/.keep
+	. $(PYTHON_ROOTFS)/bin/activate && cd $(LIBPYTHON_SHAPELY_SRC) && pip install . --prefix=$(PYTHON_ROOTFS) --no-compile
+	_dir=`find "$(PYTHON_ROOTFS)" -maxdepth 4 -type d -name shapely`; \
+	find "$$_dir" -type f -name '*.so' -delete; \
+	find "$$_dir" -type d -name '__pycache__' | xargs rm -rf; \
+	find "$$_dir" -type f | grep -v '\.py$$' | tr "\n" "\0" | xargs -0 rm; \
+	find "$(LIBPYTHON_SHAPELY_BASE)/importfix/shapely" -mindepth 1 -maxdepth 1 | xargs $(CP) -rp -t "$$_dir"; \
+	test `basename $$(dirname $$_dir) | cut -d. -f1,2` != $(LIBPYTHON3_VERSION) && \
+		mv "$$_dir" "$(PYTHON_ROOTFS)/lib/python$(LIBPYTHON3_VERSION)/shapely"
+	touch $@
+
+# Add shapely rootfs to main python
+python-rootfs: $(PYTHON_ROOTFS)/.shapely_done
```

Comment:
Who will configure `PYTHON_ROOTFS`? Will it be done by `lib-python3`?

Reply:
Yes, `PYTHON_ROOTFS` is set when the `python-rootfs` make target is being run and taken from whatever you specify on the cmdline under `path=`, with the logic being provided by lib-python3.
If lib-python3 is not present/selected this line becomes a no-op, as no recipe for python-rootfs exists, nor is it an implicit target of anything.

Reply:
Got it, nice.